crates/metastore/src/database.rs

```diff
@@ -1419,6 +1423,25 @@ impl BuiltinCatalog {
     }
 }
 
+#[cfg(test)]
+impl DatabaseCatalog {
+    /// Try to mutate the catalog and immediately commit the changes.
+    ///
+    /// Errors if the provided version doesn't match the version of the current
+    /// catalog.
+    ///
+    /// On success, a full copy of the updated catalog state will be returned.
+    // TODO: All or none.
```

Comment:
I think this comment is stale, also not sure why this is test-only.

Reply:
The reason this isnt exposed elsewhere is that the actions need to go over grpc and instead of creating 'mutate', 'mutate and commit', and 'commit' actions, all we really need on the GRPC service is 'mutate' and 'commit'. So the interfaces on the other side of things use a 'mutate_and_commit' that makes both grpc calls and only serves as a utility function. 


the same applies here, but it is on the other side of grpc, so it's not actually "used" anywhere else beyond testing. 

There were a lot of instances where this was used in testing, so it made sense to not manually rewrite all of the tests.

Reply:
it'd make sense to want to have the combined operation _behind_ the gRPC interface so that you wouldn't need to have an extra round trip in-between the operations (thereby reducing the risk of conflicts), but I am relatively unfamiliar with the code.